azure-quantum/azure/quantum/qiskit/qir/builder.py

```diff
@@ -0,0 +1,106 @@
+##
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+##
+
+from dis import Instruction
+from typing import Union, Iterable
+
+from pyqir_generator import QirBuilder
+from qiskit.circuit.classicalregister import ClassicalRegister
+from qiskit.circuit.quantumregister import QuantumRegister
+
+
+INSTRUCTIONS = [
+    "cx",
+    "cz",
+    "h",
+    "m",
+    "reset",
+    "rx",
+    "ry",
+    "rz",
+    "s",
+    "t",
+    "x",
+    "y",
+    "z"
+]
+
+
+class QiskitToQirBuilder(QirBuilder):
+    """Qiskit to QIR builder."""
+    def __init__(self, module: str):
+        """Create QiskitToQirBuilder object
+
+        :param module: QIR module name
+        :type module: str
+        """
+        super().__init__(module=module)
+
+    def add_register(self, register: Union[QuantumRegister, ClassicalRegister]):
+        """Add register
+
+        :param register: A Qiskit quantum or classical register object
+        :type register: Union[QuantumRegister, ClassicalRegister]
```

Comment:
I think Sphinx / DocFX may be able to pull this from the type hint without duplicating the type in the docstring — may be good to check.